sphinx/transforms/post_transforms/__init__.py

```diff
@@ -222,24 +218,35 @@ def has_visitor(translator: Type[nodes.NodeVisitor], node: Type[Element]) -> boo
             # subclass of SphinxTranslator supports desc_sig_element nodes automatically.
             return
 
-        if all(has_visitor(translator, node) for node in self.SIG_ELEMENTS):
-            # the translator supports all desc_sig_element nodes
-            return
-        else:
-            self.fallback()
+        # for the leaf elements (desc_sig_element), the translator should support _all_
+        if not all(has_visitor(translator, node) for node in addnodes.SIG_ELEMENTS):
+            self.fallback(addnodes.desc_sig_element)
 
-    def fallback(self) -> None:
-        for node in self.document.traverse(addnodes.desc_sig_element):
+        if not has_visitor(translator, addnodes.desc_inline):
+            self.fallback(addnodes.desc_inline)
+
+    def fallback(self, nodeType: Any) -> None:
+        for node in self.document.traverse(nodeType):
             newnode = nodes.inline()
             newnode.update_all_atts(node)
             newnode.extend(node)
             node.replace_self(newnode)
 
 
+class DescSigAddDomainAsClass(SphinxPostTransform):
```

Comment:
How about this name?
```suggestion
class PropagateDescDomain(SphinxPostTransform):
```

"propagate" is a term already used in docutils: `docutils.transforms.references.PropagateTargets`

Reply:
Good idea.